tool/planet/units.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+
+	"github.com/gravitational/planet/lib/utils"
+
+	systemdDbus "github.com/coreos/go-systemd/dbus"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+func newUnitMonitor(units ...string) (*unitMonitor, error) {
+	conn, err := systemdDbus.NewSystemConnection()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &unitMonitor{
+		conn:  conn,
+		units: units,
+	}, nil
+}
+
+type unitMonitor struct {
+	conn  *systemdDbus.Conn
+	units []string
+}
+
+func (r *unitMonitor) close() {
+	r.conn.Close()
+}
+
+func (r *unitMonitor) start(ctx context.Context) error {
+	resultC := make(chan string, len(r.units))
+	// Number of successfully started start jobs
+	var numStarted int
+	for _, unit := range r.units {
+		logger := log.WithField("unit", unit)
+		logger.Info("Start service.")
+		if _, err := r.conn.StartUnit(unit, "replace", resultC); err != nil {
+			logger.WithError(err).Warn("Failed to start unit.")
+			continue
```

Comment:
Hmm, why do we only log this error and keep going? This way the caller will still think that everything has been started successfully, no? Can we collect all errors in trace.Aggregate, return to the caller and have it retry (I think you already have retry on error logic) so it can start units that failed to start before? Otherwise this function never actually returns an error it seems like.